plugins/woocommerce/includes/admin/class-wc-admin-meta-boxes.php

```diff
@@ -226,24 +226,31 @@ public function save_meta_boxes( $post_id, $post ) {
 	}
 
 	/**
-	 * Remove block-based templates from the list of available templates for products.
+	 * Remove irrelevant block templates from the list of available templates for products.
+	 * This will also remove custom created templates.
 	 *
 	 * @param string[] $templates Array of template header names keyed by the template file name.
 	 *
 	 * @return string[] Templates array excluding block-based templates.
 	 */
-	public function remove_block_templates( $templates ) {
-		if ( count( $templates ) === 0 || ! function_exists( 'gutenberg_get_block_template' ) ) {
+	public function filter_product_block_templates( $templates ) {
```

Comment:
I'm not sure about this, but if we change the name of a public method, wouldn't it be considered a breaking change?

Reply:
You're right I think, I've changed this back.